crates/uv-trampoline/src/bounce.rs

```diff
@@ -46,21 +48,35 @@ enum TrampolineKind {
 }
 
 impl TrampolineKind {
-    const fn magic_number(&self) -> &'static [u8; 4] {
-        match self {
-            Self::Script => b"UVSC",
-            Self::Python => b"UVPY",
+    fn from_resource(data: &[u8]) -> Option<Self> {
+        match data.first() {
+            Some(1) => Some(Self::Script),
+            Some(2) => Some(Self::Python),
+            _ => None,
         }
     }
+}
 
-    fn from_buffer(buffer: &[u8]) -> Option<Self> {
-        if buffer.ends_with(Self::Script.magic_number()) {
-            Some(Self::Script)
-        } else if buffer.ends_with(Self::Python.magic_number()) {
-            Some(Self::Python)
-        } else {
-            None
-        }
+/// Safely loads a resource from the current module
+fn load_resource(resource_id: &str) -> Option<Vec<u8>> {
+    unsafe {
+        let mut resource_id_null_term = resource_id.encode_utf16().collect::<Vec<_>>();
+        resource_id_null_term.push(0);
+        // Find the resource
+        let resource = FindResourceW(
+            None,
+            windows::core::PCWSTR(resource_id_null_term.as_ptr()),
+            windows::core::PCWSTR(RT_RCDATA as *const _),
+        );
```

Comment:
Everything else in trampoline operates on utf8 and uses the `A` functions because we set a manifest declaring support for utf8 -- does that not work here?